xbmc/games/addons/input/GameClientInput.cpp

```diff
@@ -170,9 +178,14 @@ float CGameClientInput::GetPortActivation(const std::string& portAddress)
 {
   float activation = 0.0f;
 
-  auto it = m_joysticks.find(portAddress);
-  if (it != m_joysticks.end())
-    activation = it->second->GetActivation();
+  if (m_portMutex.try_lock())
+  {
+    auto it = m_joysticks.find(portAddress);
+    if (it != m_joysticks.end())
+      activation = it->second->GetActivation();
+
+    m_portMutex.unlock();
+  }
```

Comment:
Maybe use `std::unique_lock` to get rid of the plain mutex modifications? It's just a style suggestion, I don't think anything can go wrong here.

Reply:
This is the intended pattern for a try lock in the stdlib. A unique lock blocks to acquire but a try lock never blocks. Because this is called on the GUI thread we don't want to block. There isn't really an RAII pattern for try locks that hides the plain mutex modification.

Reply:
That is what `std::defer_lock` is for. It gives the ability to try_lock while still freeing the mutex in case of an exception:
```suggestion
  std::unique_lock<std::recursive_mutex> lock(m_portMutex, std::defer_lock);
  if (lock.try_lock())
  {
    auto it = m_joysticks.find(portAddress);
    if (it != m_joysticks.end())
      activation = it->second->GetActivation();
  }
```

Reply:
updated